pbkdf2/src/lib.rs

```diff
@@ -74,3 +99,140 @@ pub fn pbkdf2<F>(password: &[u8], salt: &[u8], c: usize, res: &mut [u8])
         pbkdf2_body(i, chunk, &prf, salt, c);
     }
 }
+
+/**
+ * pbkdf2_simple is a helper function that should be sufficient for the majority of cases where
+ * an application needs to use PBKDF2 to hash a password for storage. The result is a String that
+ * contains the parameters used as part of its encoding. The pbkdf2_check function may be used on
+ * a password to check if it is equal to a hashed value.
+ *
+ * # Format
+ *
+ * The format of the output is a modified version of the Modular Crypt Format that encodes algorithm
+ * used and iteration count. The format is indicated as "rpbkdf2" which is short for "Rust PBKF2
+ * format."
+ *
+ * $rpbkdf2$0$<base64(c)>$<base64(salt)>$<based64(hash)>$
+ *
+ * # Arguments
+ *
+ * * password - The password to process as a str
+ * * c - The iteration count
+ *
+ */
+#[cfg(feature="include_simple")]
+pub fn pbkdf2_simple(password: &str, c: u32) -> io::Result<String> {
+    let mut rng = OsRng::new()?;
+
+    // 128-bit salt
+    let mut salt = [0u8; 16];
+    rng.try_fill_bytes(&mut salt)?;
+
+    // 256-bit derived key
+    let mut dk = [0u8; 32];
+
+    pbkdf2::<Hmac<Sha256>>(password.as_bytes(), &salt, c as usize, &mut dk);
+
+    let mut result = "$rpbkdf2$0$".to_string();
+    let mut tmp = [0u8; 4];
+    write_u32_be(&mut tmp, c);
+    result.push_str(&base64::encode(&tmp));
+    result.push('$');
+    result.push_str(&base64::encode(&salt));
+    result.push('$');
+    result.push_str(&base64::encode(&dk));
+    result.push('$');
+
+    Ok(result)
+}
+
+/**
+ * pbkdf2_check compares a password against the result of a previous call to pbkdf2_simple and
+ * returns true if the passed in password hashes to the same value.
+ *
+ * # Arguments
+ *
+ * * password - The password to process as a str
+ * * hashed_value - A string representing a hashed password returned by pbkdf2_simple()
+ *
+ */
+#[cfg(feature="include_simple")]
+pub fn pbkdf2_check(password: &str, hashed_value: &str) -> Result<bool, &'static str> {
```

Comment:
Can you please use the same API as in the `scrypt` crate, i.e. use `Result<(), CheckError>`? In future we will probably move these function to a trait, but for now it would be nice to keep API consistent across crates.

Reply:
Done